guides/cheatsheets/git.md

```diff
@@ -0,0 +1,305 @@
+you can push multiple branches and only force one like so: (+ denotes that the
+following branch will be force pushed (issue/1))_.
+
+    git push origin +issue/1 issue/2
+
+_Unstage a file while retaining the changes in working directory_.
+
+    git reset [file]
+
+_Diff of what is changed but not staged_.
+
+    git diff
+
+_Diff of what is staged but not yet commited_.
+
+    git diff --staged
+
+---
+
+## **BRANCH & MERGE**
+
+---
+
+Isolating work in branches, changing context, and integrating changes.
+
+_List your branches. a\*will appear next to the currently active branch\._
+
+    git branch
+
+_Create a new branch_.
+
+    git branch [branch-name]
+
+_Switch to another branch and check it out into your working directory_.
+
+    git checkout
+
+_Switch to the specified branch. Creates the branch in your local repository if
+it doesn't already exist_.
+
+    git checkout -b [branch_name]
+
+_Merge the specified branch’s history into the current one_.
+
+    git merge [branch]
+
+_Show all commits in the current branch’s history_.
+
+    git log
+
+---
+
+## **INSPECT & COMPARE**
+
+---
+
+Examining logs, diffs and object information.
+
+_Show the commit history for the currently active branch_.
+
+    git log
+
+_Show the commits on branchA that are not on branchB_.
+
+    git log branchB..branchA
+
+_Show the commits that changed file, even across renames_.
+
+    git log --follow [file]
+
+_Get the author of a commit_.
+
+    git blame [commit_hash]
+
+_Show the diff of what is in branchA that is not in branchB_.
+
+    git diff branchB...branchA
+
+_Show any object in Git in human-readable format_.
+
+    git show [SHA]
+
+---
+
+## **TRACKING PATH CHANGES**
+
+---
+
+Versioning file removes and path changes.
+
+_Delete the file from project and stage the removal for commit_.
+
+    git rm [file]
+
+_Change an existing file path and stage the move_.
+
+    git mv [existing-path] [new-path]
+
+_Show all commit logs with indication of any paths that moved_.
+
+    git log --stat -M
+
+---
+
+## **IGNORING PATTERNS**
+
+---
+
+Preventing unintentional staging or committing of files.
+
+_Save a file with desired patterns as .gitignore with either direct string
+matches or wildcard globs_.
+
+    logs/
+    *.notes
+    pattern*/
+
+_System wide ignore patern for all local repositories_.
+
+    git config --global core.excludesfile [file]
+
+---
+
+## **SHARE & UPDATE**
+
+---
+
+Retrieving updates from another repository and updating local repos.
+
+_Add a git URL as an alias_.
+
+    git remote add [alias] [url]
+
+_Fetch down all the branches from that Git remote_.
+
+    git fetch [alias]
+
+_Merge a remote branch into your current branch to bring it up to date_.
+
+    git merge [alias]/[branch]
+
+_Transmit local branch commits to the remote repository branch_.
+
+    git push [alias] [branch]
+
+_Fetch and merge any commits from the tracking remote branch_.
+
+    git pull
```

Comment:
This will only work if remote-tracking is correctly set up, otherwise it'll need to be the more explicit `git pull [remote] [branch]`.